schema/parser.cpp

```diff
@@ -1340,11 +1340,11 @@ bool Parser::resolveForwardDeclarations()
                 }
             }
         }
-        foreach (const AttributeGroup &group, complexType.attributeGroups()) {
+        for (const AttributeGroup &group : complexType.attributeGroups()) {
```

Comment:
needs local var

(well all this is for performance, not correctness. I can accept the answer that we don't care that much about performance in this code --- but the request for adding qAsConst in the first place was for performance reasons too ;)

Of course the decision can be "qAsConst on named vars, detach/deepcopy on temporaries", for readability reasons...

Reply:
Well if my opinion counts: we should care about performance: there are quite big XML/XSD/WSDL files out in the wild.